### DISCUSSION TITLE
"transition_finished" signal error

### DISCUSSION THREAD

**Discussion by reikoanu** [dialogic-godot/dialogic#2098] [Q&A]
Hello! I'm pretty sure this is a minor error, as I think the _on_transition_finished function still transitions to the next scene and the game works fine otherwise, but it's still something interesting that I've been trying to figure out.

so basically every timeline emits a signal before ending, and I have an `_on_dialogic _signal` that listens to all of them and most of them call `get_tree().change_scene_to_file("blahblah")`, but every time that timelines emit a signal I get this error, even if the signal is just supposed to print something

``` emit_signalp: Error calling from signal 'transition_finished' to callable: 'Node(subsystem_backgrounds.gd)::_on_transition_finished': Cannot convert argument 1 from Object to Object.```

a friend suggested to me that it might have something to do with animations or transitions either in the scenes I change to (or I think it's something about the dialogic timeline I'm not implementing correctly). Only one of my scene changes doesn't get this error and it is a scene change back to the main menu.

I do notice that before every one of these errors, I get this warning right before (and according to the debug timer it is immediately followed by the above error) (EDIT: this is not necessarily true for every error, but most of them)

``` start: Target object freed before starting, aborting Tweener.```

Any insight into what is causing this error, and if there are any fixes, or if this is a dialogic bug, would be very helpful.
Thank you in advance!

**Comment by Jowan-Spooner** [comment]
Hmm... The transition_finished signal is part of the Backgrounds system. Is it likely that a background might still be transitioning while you change the scene? Maybe that produces this error. 

I will try to recreate this. Thanks for the report.

**Reply by Jowan-Spooner** [reply]
I might have a fix for this. It would be easiest if we could write on discord.

**Reply by reikoanu** [reply]
Oh thank you for replying so quickly, and sorry for replying a little late. Would I be able to reach out to you through the discord server?
